### DISCUSSION TITLE
Ring Alarm State

### DISCUSSION THREAD

**Discussion by Anthael** [dgreif/ring#1087] [Q&A]
Hi

I'm running the last version v11.3.1, and basec from the example.ts, I m trying to get the alarm mode working.
From the api I found the property LocationModeInput / LocationMode, but I can not retrieve the value.

From the beginning I guest the value was set directly from the base station but the value seems to be set on the Location.

I have tried to expand the different values from the objects to identify the current alarm mode but impossible.

Do you have an example with the alarm mode ? or how did you get it ?

Thanks by advance

**Comment by dgreif** [comment]
You can _set_ the mode directly on the location with [these functions](https://github.com/dgreif/ring/blob/ea6f6aaef3a32d9a79ee581073d1b2755cca6fa5/api/location.ts#L415-L425). To get the current mode, you have to grab the [security panel device](https://github.com/dgreif/ring/blob/ea6f6aaef3a32d9a79ee581073d1b2755cca6fa5/api/location.ts#L396-L413) and look at `device.data.mode` which will be `all` (away), `some` (stay) or `none (disarmed).
